### PR TITLE
Implemented layer structure with Delivery, UseCases, and Repository

### PR DESCRIPTION
**Pull request by SpanFish11**

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import express from "express";
+import config from "./config/config.js";
+import meteorRouter from "./delivery/meteor_routes.js";
+
+const app = express();
+const port = config.port;
+
+app.use(express.json());
+app.use(meteorRouter);
+
```
Comment (Mikhail1992) [inline app/app.js:10]: add `errorHandler` please somewhere there

here you can find a little bit more information https://medium.com/@mohsinogen/simplified-guide-to-setting-up-a-global-error-handler-in-express-js-daf8dd640b69

File: app/repository/meteors_repository.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+                start_date: config.startDate,
+                end_date: config.endDate,
+                api_key: config.apiKey,
+            }
+        });
+
+        return response.data.near_earth_objects;
+    } catch (error) {
+        throw new Error('Error fetching meteor data from repository');
+    }
```
Comment (Mikhail1992) [inline app/repository/meteors_repository.js:17]: You can remove try..catch here and leave it only in the controller

File: app/delivery/meteor_routes.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+meteorRouter.get('/meteors', async (req, res) => {
+    try {
+        const meteors = await getMeteors();
+
+        res.json(meteors);
+    } catch (error) {
+        console.error('Error in delivery layer:', error);
+
+        res.status(500).json({error: 'An error occurred while fetching data.'});
```
Comment (Mikhail1992) [inline app/delivery/meteor_routes.js:14]:
```
next(error)
```
Error - can be a custom object
Example
```js
class Exception extends Error {
  constructor(code, massage) {
    this.code = code;
    this.massage = massage;
  }
}

...

next(new Exception(500, 'Error message')
```

**Review by Mikhail1992 — COMMENTED** [review]
Looks good
Just a couple of comments

File: app/app.js (hunk 1 above)
Comment (SpanFish11) [reply app/app.js:12]: Done

File: app/repository/meteors_repository.js (hunk 2 above)
Comment (SpanFish11) [reply app/repository/meteors_repository.js]: done

File: app/delivery/meteor_routes.js (hunk 3 above)
Comment (SpanFish11) [reply app/delivery/meteor_routes.js]: done

File: app/delivery/meteor_routes.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+const meteorRouter = express.Router();
+
+meteorRouter.get('/meteors', async (req, res, next) => {
+    try {
+        const meteors = await getMeteors();
+
+        res.json(meteors);
+    } catch (error) {
+        const status = error.status || 500;
```
Comment (Mikhail1992) [inline app/delivery/meteor_routes.js:13]: you already have this line here https://github.com/SpanFish11/node-proxy-server/pull/7/files#diff-f35672458e83e5dbbb2cdd7d523de438dc10d6a30272994170e6558dd7c8860fR4

File: app/delivery/meteor_routes.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+        next(new Exception(status, `An error occurred while fetching data: ${error.message}`));
```
Comment (Mikhail1992) [inline app/delivery/meteor_routes.js:15]: ext(new Exception(error.code, `An error occurred while fetching data: ${error.message}`));

**Review by Mikhail1992 — APPROVED** [review]

**Pull request merged**